services/order.py

```diff
@@ -0,0 +1,33 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, MovieSession, Ticket
+
+
+@transaction.atomic
+def create_order(tickets: list[dict], username: str, date: str = None) -> None:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+    if date:
+        order.created_at = date
+
+    for ticket in tickets:
+        movie_session = MovieSession.objects.get(pk=ticket["movie_session"])
+        row = ticket["row"]
+        seat = ticket["seat"]
+        Ticket.objects.create(
+            movie_session=movie_session,
+            order=order,
+            seat=seat,
+            row=row,
+        )
+
+        order.save()
```

Comment:
You are calling the `save()` method inside the loop, which violates the 'Don't Repeat Yourself' principle from the checklist. Consider moving `order.save()` outside the loop to call it only once after all tickets have been processed.